addons/mrp/views/mrp_production_views.xml

```diff
@@ -316,7 +316,7 @@
                                     readonly="state != 'draft'"/>
                                 <button name="action_generate_bom" type="object" icon="fa-plus"
                                     title="Generate a new BoM from this Manufacturing Order" groups="mrp.group_mrp_manager"
-                                    invisible="bom_id or not product_id or not move_raw_ids">
+                                    invisible="bom_id or not product_id or not move_raw_ids or not show_generate_bom">
```

Comment:
Move the entire condition to `show_generate_bom`'s compute. No point in splitting it in two different places. :wink: